src/pokemon/models.ts

```diff
@@ -0,0 +1,105 @@
+export interface Dex {
+	pokemon: Pokemon[];
+	formats: Format[];
+	natures: Nature[];
+	abilities: Ability[];
+	moveflags: Moveflag[];
+	moves: Move[];
+	types: Type[];
+	items: Item[];
+}
+
+interface BaseType {
+	name: string;
+	description: string;
+	genfamily: string[];
+}
+
+interface Type extends BaseType {
+	atk_effectives: (number | string)[][];
+}
+  
+interface Move extends BaseType {
+	cap: boolean;
+	category: string;
+	power: number;
+	accuracy: number;
+	priority: number;
+	pp: number;
+	type: string;
+	flags: string[];
+}
+  
+interface Moveflag extends BaseType {
+}
+  
+interface Ability extends BaseType {
+	cap: boolean;
+}
+
+interface Item extends BaseType {
+	cap: boolean;
+}
+  
+interface Nature {
+	name: string;
+	hp: number;
+	atk: number;
+	def: number;
+	spa: number;
+	spd: number;
+	spe: number;
+	summary: string;
+	genfamily: string[];
+}
+  
+interface Format {
+	name: string;
+	shorthand: string;
+	genfamily: string[];
+}
+
+export interface Pokemon {
+	name: string;
+	hp: number;
+	atk: number;
+	def: number;
+	spa: number;
+	spd: number;
+	spe: number;
+	weight: number;
+	height: number;
+	types: PokemonType[];
+	abilities: string[];
+	formats: string[];
+	oob?: Oob;
+}
+  
+interface Oob {
```

Comment:
Oob?

Reply:
It's a metadata from Smogon Dex. Please see an example:
```
"oob":{"dex_number":63,"cap":false,"evos":["Kadabra"],"alts":[],"genfamily":["RB","GS","RS","DP","BW","XY","SM"]}}
```